climpred/prediction.py

```diff
@@ -838,12 +839,35 @@ def compute_persistence_pm(ds, control, nlags, metric='pearson_r',
             'rmse',
             'mse',
             'mae'""")
+
+    init_years = ds['initialization'].values
+    if isinstance(ds.time.values[0], cftime._cftime.DatetimeProlepticGregorian) or isinstance(ds.time.values[0], np.datetime64):
+        init_cftimes = []
+        for year in init_years:
+            init_cftimes.append(control.sel(
+                time=str(year)).isel(time=init_month_index).time)
+        init_cftimes = xr.concat(init_cftimes, 'time')
+    elif isinstance(ds.time.values[0], np.int64):
+        init_cftimes = []
+        for year in init_years:
+            init_cftimes.append(control.sel(
+                time=year).time)
+        init_cftimes = xr.concat(init_cftimes, 'time')
+    else:
+        raise ValueError(
+            'Set time axis to xr.cftime_range, pd.date_range or np.int64.')
+
+    inits_index = []
+    control_time_list = list(control.time.values)
+    for i, inits in enumerate(init_cftimes.time.values):
```

Comment:
Another note is that some of this might need to change slightly to comply with my fixes mentioned in https://github.com/bradyrx/climpred/issues/85. Just make sure that my snippet of code for DPLE in that issue thread runs properly with these time updates.